src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -761,6 +761,8 @@ await scheduler.QueueTask(this.membershipOracle.ShutDown, this.membershipOracleC
                         .WithCancellation(ct, "MembershipOracle Shutting down failed because the task was cancelled");
                     // Deactivate all grains
                     SafeExecute(() => catalog.DeactivateAllActivations().Wait(ct));
+                    //wait for all queued message sent to OutboundMessageQueue before MessageCenter stop and OutboundMessageQueue stop. 
+                    await Task.Delay(TimeSpan.FromSeconds(2));
```

Comment:
Why hardcoded 2 seconds? Don't we want to wait here for forwarding of queued messages?

Reply:
this is just to wait for messages to be enqueued into OutboundMessageQueue, not actually forwarding yet. So 2 sec should be enough. What should we do in your opinion>?

Reply:
Doesn't `catalog.DeactivateAllActivations().Wait(ct)` return only after all activations got deactivated and their messages got enqueued for forwarding?